src/game.cpp

```diff
@@ -12965,6 +12965,12 @@ void game::display_om_pathfinding_progress( size_t /* open_set */, size_t /* kno
     inp_mngr.pump_events();
 }
 
+void game::wait_popup_reset()
+{
+    wait_popup.reset();
+    ui_manager::redraw();
```

Comment:
Is this line necessary for the wait popup to disappear? I think `wait_popup.reset()` will call `ui_adaptor`'s deconstruction function which should mark the underlying UI for redraw, as long as the `ui_adaptor`'s dimensions are correctly set, and the underlying UI will call `ui_manager::redraw` before checking for the next input, so this line is probably not needed if everything works as intended.